### DISCUSSION TITLE
New field type: Editor

### DISCUSSION THREAD

**Discussion by apexdodge** [RaythaHQ/raytha#16] [Ideas]
We currently use the [Trix Editor](https://github.com/basecamp/trix) by Basecamp, which is a very effective editor for writing and blogging. However, it lacks advanced features that may be desired by some users. Some of the limitations of the Trix Editor include:

- Lack of ability to modify the dimensions of image uploads
- Links cannot be set to open in new tabs
- No table support
- Limited options for font, font size, background color, etc.
- Cannot embed videos

To address these limitations, we propose adding an additional field type called "Editor" or "WYSIWYG" to our system. This field type would include the full suite of features that users expect in an editor, such as the ability to modify image dimensions, create tables, and embed videos. System admins would still have the option to use the Long Text field type, but the Editor field type would provide additional functionality for those who need it.

As a side note, we were previously unaware that the TinyMCE editor is licensed under the MIT license. This could potentially be an alternative option to consider as well.

**Comment by fasteddys** [comment]
Lets break it down to text, and then the over structure of the page and content types
**for text:** 

- [ ] https://quilljs.com/,
- [ ] https://summernote.org/
- [ ] https://alloyeditor.com/

**Content/Wix like:**

- [ ] Check out `vvveb.js` or `grape.js` here on github
- [ ] This goes beyond the text, and allows you to do something like wp elementor or divi or **wix**, inside your pages

**Comment by apexdodge** [comment]
Yes, the text editor would be done first but I've done extensive research into wysiwygs and TinyMCE https://github.com/tinymce/tinymce is the best option among MIT licensed offerings. The good news is that I think we might be able to do add this feature without a db migration so it will be among the first items on the next milestone after the 1.0 release.

**Reply by fasteddys** [reply]
just watchout for their license, very restrictive, quill is used behind many of the bigger ones. but if you have done your homework thats awesome!!
